BoostTestAdapter/Utility/VisualStudio/DefaultBoostTestPackageServiceFactory.cs

```diff
@@ -18,21 +19,20 @@ class DefaultBoostTestPackageServiceFactory : IBoostTestPackageServiceFactory
 
         public IBoostTestPackageServiceWrapper Create()
         {
-            // Assuming the adapter is running in a child process of the Visual Studio process.
             int processId = Process.GetCurrentProcess().Id;
-            int parentProcessId = GetParentProcessId(processId);
+            int parentProcessId = GetParentVSProcessId(processId);
```

Comment:
So we need the PID just to construct the pipe name to connect to the right instance of VS, right? Perhaps we should make it more robust by implementing the `IRunSettingsService` and communicating the pipe name from the VS package side to the adapter side (like TAfGT is doing)?

Reply:
That's correct. I'll take a look IRunSettingsService, that does seem like a nicer solution.

Reply:
Using this now!